src/firefly/js/templates/lightcurve/LcPhaseTable.js

```diff
@@ -0,0 +1,232 @@
+import {dispatchTableReplace} from '../../tables/TablesCntlr.js';
+import {LC, getValidValueFrom} from './LcManager.js';
+import {getLayouInfo} from '../../core/LayoutCntlr.js';
+import FieldGroupUtils from '../../fieldGroup/FieldGroupUtils';
+
+
+const DEC_PHASE = 3;       // decimal digit
+
+/**
+ * @summary upload phase table with the column arranged as needed
+ * @param {TableModel} tbl
+ * @param {string} flux
+ */
+export function uploadPhaseTable(tbl, flux) {
+    const {tableData, tbl_id, tableMeta, title} = tbl;
+
+    const ipacTable = tableToText(tableData.columns, tableData.data, true, tableMeta);
+    const file = new File([new Blob([ipacTable])], `${tbl_id}.tbl`);
+
+    doUpload(file).then(({status, cacheKey}) => {
+        const tReq = makeFileRequest(title, cacheKey, null, {tbl_id, sortInfo:sortInfoString(LC.PHASE_CNAME)});
+        dispatchTableSearch(tReq, {removable: true});
+
+        const xyPlotParams = {
+            userSetBoundaries: {xMax: 2},
+            x: {columnOrExpr: LC.PHASE_CNAME, options: 'grid'},
+            y: {columnOrExpr: flux, options: 'grid,flip'}
+        };
+        loadXYPlot({chartId: tbl_id, tblId: tbl_id, xyPlotParams});
+    });
+
+}
+
+/**
+ * @summary adding phase column to raw table
+ * @param {string} flux flux column name
+ * @param {string} time time column name
+ * @param {string} period periodt  value
+ * @param {string} tzero zero time value
+ * @return {TableModel} phase table
+ */
+export function doPFCalculate(flux, time, period, tzero) {
+    const rawTable = getTblById(LC.RAW_TABLE);
+
+    var phaseFoldedTable = rawTable && addPhaseToTable(rawTable, time, flux, tzero, period);
+    phaseFoldedTable&&repeatDataCycle(time, parseFloat(period), phaseFoldedTable);
+
+    return phaseFoldedTable;
+}
+
+
+/**
+ * @summary calculate phase and return as text
+ * @param {number} time
+ * @param {number} timeZero
+ * @param {number} period
+ * @param {number} dec
+ * @returns {string}  folded phase (positive or negative) with 'dec'  decimal digits
+ */
+export function getPhase(time, timeZero, period,  dec=DEC_PHASE) {
+    var q = (time - timeZero)/period;
+    var p = q >= 0  ? (q - Math.floor(q)) : (q + Math.floor(-q));
+
+    return p.toFixed(dec);
+}
+
+/**
+ * @summary create a table model with phase column
+ * @param {TableModel} tbl
+ * @param {string} timeName
+ * @param {string} fluxName
+ * @param {string} tzero
+ * @param {string} period
+ * @returns {TableModel}
+ */
+function addPhaseToTable(tbl, timeName, fluxName, tzero, period) {
+    var {columns} = tbl.tableData;
+    var tIdx = timeName ? columns.findIndex((c) => (c.name === timeName)) : -1;
+
+    if (tIdx < 0) return null;
+
+    var tPF = Object.assign(cloneDeep(tbl), {tbl_id: LC.PHASE_FOLDED, title: 'Phase Folded'},
+                                            {request: getPhaseFoldingRequest(period, timeName, fluxName, tbl)},
+                                            {highlightedRow: 0});
```

Comment:
I think you do.  The resulting table will have more columns and twice as many rows as the original.